package.json

```diff
@@ -1,49 +1,240 @@
 {
-    "name": "yarn-spinner",
-    "displayName": "Yarn Spinner",
-    "description": "Adds support for the Yarn Spinner dialogue language.",
-    "version": "1.1.0",
-    "homepage": "https://yarnspinner.dev",
-    
-    "license": "MIT",
-    "keywords": [
-        "yarn spinner", "yarnspinner", "yarn", "game writing", "narrative", "tree", "branch", "branching", "dialog", "dialogue"
+  "name": "yarn-spinner",
+  "displayName": "Yarn Spinner",
+  "description": "Adds support for the Yarn Spinner dialogue language.",
+  "version": "1.1.0",
+  "homepage": "https://yarnspinner.dev",
+  "license": "MIT",
+  "keywords": [
+    "yarn spinner",
+    "yarnspinner",
+    "yarn",
+    "game writing",
+    "narrative",
+    "tree",
+    "branch",
+    "branching",
+    "dialog",
+    "dialogue"
+  ],
+  "qna": false,
+  "publisher": "SecretLab",
+  "icon": "icon.png",
+  "galleryBanner": {
+    "color": "#C5E29B",
+    "theme": "light"
+  },
+  "engines": {
+    "vscode": "^1.44.0"
+  },
+  "categories": [
+    "Programming Languages"
+  ],
+  "activationEvents": [
+    "onCommand:yarnSpinner.start",
+    "onCustomEditor:yarnSpinner.editor"
+  ],
+  "main": "./out/extension.js",
+  "contributes": {
+    "languages": [
+      {
+        "id": "yarnspinner",
+        "aliases": [
+          "Yarn Spinner",
+          "yarnspinner",
+          "Yarn"
+        ],
+        "extensions": [
+          ".yarn",
+          ".yarn.txt"
+        ],
+        "configuration": "./language-configuration.json"
+      }
     ],
-    "qna": false,
-    "publisher": "SecretLab",
-    "icon": "icon.png",
-    "galleryBanner": {
-        "color": "#C5E29B",
-        "theme": "light"
-    },
-    "engines": {
-        "vscode": "^1.38.0"
-    },
-    "categories": [
-        "Programming Languages"
+    "grammars": [
+      {
+        "language": "yarnspinner",
+        "scopeName": "source.yarnspinner",
+        "path": "./syntaxes/yarnspinner.tmLanguage.json"
+      }
     ],
-    "contributes": {
-        "languages": [
-            {
-                "id": "yarnspinner",
-                "aliases": [
-                    "Yarn Spinner",
-                    "yarnspinner",
-                    "Yarn"
-                ],
-                "extensions": [
-                    ".yarn",
-                    ".yarn.txt"
-                ],
-                "configuration": "./language-configuration.json"
-            }
+    "customEditors": [
+      {
+        "viewType": "yarnSpinner.editor",
+        "displayName": "Yarn Editor",
+        "selector": [
+          {
+            "filenamePattern": "*.yarn"
```

Comment:
I considered also adding `.json`, `.twee`, and all the other supported formats with `"priority": "optional"` here but I'm not sure how supported those are in the editor.